docs/README.md

```diff
@@ -676,15 +676,112 @@ In this chapter you'll walk through these concepts and its details.
 
 ### 2. Events
 
+Events are **immutable facts** within your application's domain. They are the cornerstone in Booster because of its event-driven and event-sourced nature. Booster events are TypeScript classes decorated with `@Event `:
+
+```typescript
+@Event
+export class EventName {
+  public constructor(readonly field1: SomeType,
```

Comment:
one of these fields must be the id of the entity, it could be of type UUID or string, is it worth doing something in relation to that?

Reply:
Yes @tainguyenbui, good catch I'll change it

Reply:
@tainguyenbui, regarding that restriction, is it actually needed to inject the entity ID in the event constructor directly? Or is it ok to just be able to find the entity ID within one of the arguments?

I mean, can we:

```typescript
class SomeType {
  constructor(readonly entityID: UUID, someField: string){}
}

//

@Event
export class EventName {
  public constructor(readonly field1: SomeType,
                     readonly field2: SomeOtherType,
                     /*as many fields as you needed*/) {}

  public entityID(): UUID {
    return field1.entityID
  }
}


```

Reply:
this id is really up to you. I wouldn't name the property entityID though.

Let's say that there is a class that has 4 properties, two of them UUID, you will then choose which one to use. That ID is important because for instance, when using read models, it will be the ID you have to pass if you want to find an item in the database

Reply:
ok, so it has nothing to do with the constructor parameters but with being able to return an existing entity ID from the `entityID` method right?

Reply:
nothing to do with the constructor, that is right

Reply:
`entityID` is a method to allow people to find their entityID in any way they want, it's not a requirement that it is a field in the event itself.

Reply:
To elaborate even more: 
There are events that represent the creation of the entity, they are the first event for a specific entity. For example, one called `CartCreated` or `ProductCreated`.

In those cases, the user could even do the following:

```typescript
@Event
export class EventName {
  public constructor(readonly field1: SomeType,
                     readonly field2: SomeOtherType,
                     /*as many fields as you needed*/) {}

  public entityID(): UUID {
    return UUID.generate()
  }
}
```

So in this case, the entityID is not even taken from any place, it is generated in place.

Because of all this variability, we made the `entityID()` member a function, instead of a field.

And also this is the reason why Events are not generated with an `ID` field by default when using generators

Reply:
Awesome feedback @alvaroloes I think that was the missing piece here 👍

Reply:
I wouldn't generate the ID value in the `entityID` function, because in that way the event can't be replayed. I rather generate the IDs before registering them in the command or event handler.

Reply:
@itrion 
@javiertoledo is right. And not only that, this would make the current Booster fail, as when we deserialize the event, we call several times to `entityID()` and we expect it to be always the same, and it would not if we generate a random string inside the function. So this is something we not only should not mention in the documentation, but we should discourage it!

Reply:
That makes total sense and it worth it to mention that the `entityID` method must also be pure then